api/mesh/v1alpha1/health_check.proto

```diff
@@ -39,6 +40,58 @@ message HealthCheck {
 
     // Number of consecutive healthy checks before considering a host healthy.
     uint32 healthy_threshold = 4 [ (validate.rules).uint32 = {gt : 0} ];
+
+    // Tcp defines optional configuration for specifying bytes to send and
+    // expected response during the health check
+    message Tcp {
+      // Bytes which will be send during the health check to the target
+      google.protobuf.BytesValue send = 1;
+
+      // Bytes blocks expected as a response. When checking the response,
+      // “fuzzy” matching is performed such that each block must be found, and
+      // in the order specified, but not necessarily contiguous.
+      repeated google.protobuf.BytesValue receive = 2;
+    }
+
+    Tcp tcp = 5;
+
+    // Http defines optional Http configuration which will instruct the service
+    // the health check will be made for is an http service. It's mutually
+    // exclusive with the Tcp block so when provided you can't provide
+    // the Tcp configuration
+    message Http {
+      // The HTTP path which will be requested during the health check
+      // (ie. /health)
+      google.protobuf.StringValue path = 1; // (REQUIRED)
+
+      message HeaderValue {
+        google.protobuf.StringValue key = 1; // (REQUIRED)
+        google.protobuf.StringValue value = 2;
+      }
+
+      message HeaderValueOption {
+        // Key/Value representation of the HTTP header
+        HeaderValue header = 1; // (REQUIRED)
+        // The bool value which if true (default) will mean the header values
+        // should be appended to already present ones
+        google.protobuf.BoolValue append = 2;
+      }
+
+      // The list of HTTP headers which should be added to each health check
+      // request
+      repeated HeaderValueOption request_headers_to_add = 2;
+
+      // List of HTTP response statuses which are considered healthy
+      repeated google.protobuf.UInt32Value expected_statuses = 3
+          [ (validate.rules).uint32 = {gte : 100, lt : 600} ];
+
+      // By default HTTP health checks will be executed using HTTP 2 protocol,
+      // if this flag will be set, HTTP 1 will be used instead
+      bool use_http1 = 4;
```

Comment:
in what case we would need to set this to true? I think we have full knowledge if it is HTTP 1 or 2 from the code.

Reply:
I thought it could add asome value to the user for being able to use HTTP 1 when needed, but don't see any immidiate benefits, so I can remove it :-)

Reply:
Removed as suggested